### PR TITLE
test for suggest-psp

### PR DESCRIPTION
**Pull request by Shell32-Natsu**

### CODE REVIEW

File: tests/e2e-test/internal/runner/util.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func gitCommit(d, msg string) error {
 	if err != nil {
 		return fmt.Errorf("git config error: %w, output: %s", err, o)
 	}
-	o, err = runCommand(d, "git", []string{"commit", "-m", msg})
+	o, err = runCommand(d, "git", []string{"commit", "-m", msg, "--allow-empty"})
```
Comment (mengqiy) [inline tests/e2e-test/internal/runner/util.go:45]: In what scenario, there will be no diff?

Comment (Shell32-Natsu) [reply tests/e2e-test/internal/runner/util.go:45]: I am not familiar with this function. But if a function is used to validate the resources then there might be no diff.

Comment (mengqiy) [reply tests/e2e-test/internal/runner/util.go:45]: When the hydration pipeline work is done, ideally, this function can be specified in the validation section and then the test harness doesn't need to do `--allow-empty`.
We can improve that later.

**Review by mengqiy — APPROVED** [review]

**Comment by mengqiy** [comment]
The test is failing with `Error:  Error: Helm command template extra-args /source --values /source/values-production.yaml results in error: Error: open /source/values-production.yaml: no such file or directory`.
I thought this was fixed in the other PR.

**Comment by Shell32-Natsu** [comment]
Yes I need to rebase

File: tests/e2e-test/internal/runner/util.go (hunk 1 above)
Comment (Shell32-Natsu) [reply tests/e2e-test/internal/runner/util.go:45]: Another possibility is the transformer is conditional.

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**